lecture-notes/l1-intro-new.md

```
@@ -0,0 +1,439 @@
+        return "Light blinks at {}% power".format(device['power'])
+    elif isFan(device):
+        return "Fan spins at {}% speed".format(device['speed'])
+    else:
+        raise ValueError("Unknown device or not a valid IoT device")
+
+```
+
+ 
+We can use these functions as follows: calling `identify(livingRoomLight)` produces the string "Light blinks at 100% power", and calling `identify(fan)` produces the string "Fan spins at 50% speed" 
+ 
+## 2.2 Extending the design
```

markdownlint-cli2 (0.22.0)

[warning] Heading style
Expected: setext; Actual: atx

(MD003, heading-style)